clayer/pysertype.c

```diff
@@ -120,47 +119,60 @@ static ddspy_serdata_t *ddspy_serdata_new(const struct ddsi_sertype* type, enum
     new->data_size = data_size;
     new->key = NULL;
     new->key_size = 0;
-    new->key_populated = false;
     new->data_is_key = false;
     new->is_v2 = ((ddspy_sertype_t*)type)->is_v2_by_default;
 
     return new;
 }
 
-static void ddspy_serdata_populate_key(ddspy_serdata_t* this)
-{
-    if (csertype(this)->keyless) {
-        this->key = NULL;
-        this->key_size = 0;
-        this->key_populated = true;
+static bool ddspy_serdata_populate_key(ddspy_serdata_t* this)
+{
+    const uint32_t xcdr_version = this->is_v2 ? DDSI_RTPS_CDR_ENC_VERSION_2 : DDSI_RTPS_CDR_ENC_VERSION_1;
```

Comment:
Looks like the keyless variant is not handled correctly, that results in a key with only the CDR header.